### DISCUSSION TITLE
Can I prevent a deploy script to run if the build script output has not changed?

### DISCUSSION THREAD

**Discussion by damien-monni** [vercel/turborepo#1245] [Help] [answered]
Let's say I have a simple next.js app with 2 scripts: build and deploy.
I only want to deploy when the build output has changed.

I tried to add dependencies to the turbo.json file but deploy is always run, no matter if build output is coming from the cache or not.

I tried:

```
{
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**", ".next/**"]
    },
    "deploy": {
      "dependsOn": ["^build"],
      "inputs": [".next/**"],
      "outputs": [".next/**"]
    }
  }
}

```

Is it possible to avoid deploying when nothing has changed?

**Comment by anthonyshew** [comment, marked as answer]
`turbo-ignore` would likely be of use here!

https://turbo.build/repo/docs/core-concepts/monorepos/skipping-tasks
